### PR TITLE
[ORC-7607][1/N] add command skeleton for flink connection

### PR DESCRIPTION
**Pull request by lihaosky**
Release Notes
-------------
New Features
* Connection management under Flink

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----
This PR adds connection command skeleton under Flink

References
----------

Test & Review
-------------

### CODE REVIEW

File: internal/flink/command_connection.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
+package flink
+
+import (
+	"github.com/spf13/cobra"
+	"strings"
```
Comment (brianstrauch) [inline internal/flink/command_connection.go:5]: Run `make lint`

File: internal/flink/command_connection.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	return cmd
+}
+
+func (c *command) validConnectionArgs(cmd *cobra.Command, args []string) []string {
+	if len(args) > 0 {
+		return nil
+	}
+
+	// TODO
```
Comment (brianstrauch) [inline internal/flink/command_connection.go:51]: Did you mean to mark this PR as "Draft" or do you want a review?

Comment (lihaosky) [reply internal/flink/command_connection.go:51]: I want a review. I want to check in the command skeleton first and then add the functionalities.

Comment (brianstrauch) [reply internal/flink/command_connection.go:52]: Can you merge this into a non-`main` branch, then?

Comment (brianstrauch) [reply internal/flink/command_connection.go:52]: We want to have the flexibility to do a release before you finish.

Comment (lihaosky) [reply internal/flink/command_connection.go:52]: This is guarded by a LD flag. Will this hide it from prod release?

Comment (brianstrauch) [reply internal/flink/command_connection.go:52]: Yes, but what's the point of releasing a half-implemented feature that's turned off?

Comment (lihaosky) [reply internal/flink/command_connection.go:52]: I'm new to CLI implementation. Do we always implement everything in one PR and merge one huge PR for a new feature? I would like to have smaller PRs and iterate on it. I guess I can create small PRs to non-main branch and you can review it and in the end we just merge all of it to main.

Comment (brianstrauch) [reply internal/flink/command_connection.go:52]: > I guess I can create small PRs to non-main branch and you can review it and in the end we just merge all of it to main.

This is the preferred way! I would also recommend doing all of this in the airlock repo so you don't have to do stuff like this:

> Use airlock to push dummy commit to a branch to create it?

Comment (lihaosky) [reply internal/flink/command_connection.go:52]: But airlock repo doesn't CI to run tests

Comment (brianstrauch) [reply internal/flink/command_connection.go:52]: You can run `make test` and `make lint` locally, too!

Comment (lihaosky) [reply internal/flink/command_connection.go:52]: Sounds good. I'll create PRs in airlock branch and you can review there

**Comment by brianstrauch** [comment]
Closing based on discussion here: https://github.com/confluentinc/cli/pull/2865#discussion_r1737286418

**Pull request closed without merging**